integ/src/updater.rs

```diff
@@ -2,52 +2,31 @@
   updater helps running brupop on existing EKS cluster and clean up all
   resources once completing integration test
 !*/
-
-use lazy_static::lazy_static;
-
 use snafu::{ensure, ResultExt};
 use std::process::Command;
 
 use k8s_openapi::api::core::v1::Node;
 use kube::api::{Api, ListParams};
 
-use models::constants::NAMESPACE;
-
 const CURRENT_LOCATION_PATH: &str = "integ/src";
 const PODS_TEMPLATE: &str = "pods-template.yaml";
 const KUBECTL_BINARY: &str = "kubectl";
 
-lazy_static! {
-    static ref BRUPOP_CLUSTER_ROLES: Vec<&'static str> = {
-        let mut m = Vec::new();
-        m.push("brupop-apiserver-role");
-        m.push("brupop-agent-role");
-        m.push("brupop-controller-role");
-        m
-    };
-}
-
-lazy_static! {
-    static ref BRUPOP_CLUSTER_ROLE_BINDINGS: Vec<&'static str> = {
-        let mut m = Vec::new();
-        m.push("brupop-apiserver-role-binding");
-        m.push("brupop-agent-role-binding");
-        m.push("brupop-controller-role-binding");
-        m
-    };
-}
-
 #[derive(strum_macros::Display, Debug)]
 pub enum Action {
-    Create,
+    Apply,
     Delete,
 }
 
-// installing brupop on EKS cluster
-pub async fn run_brupop(kube_config_path: &str) -> UpdaterResult<()> {
+// =^..^=   =^..^=   =^..^=   =^..^=   =^..^= Deletion and Creation of brupop resources  =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
+
+// install or destroy all brupop resources
+pub async fn process_brupop_resources(action: Action, kube_config_path: &str) -> UpdaterResult<()> {
+    let action_string: String = action.to_string();
+
     let brupop_resource_status = Command::new(KUBECTL_BINARY)
         .args([
-            "apply",
+            &action_string.to_lowercase(),
```

Comment:
How do you plan to handle deletion failure? I know sometimes the deletion get stuck and need to manually patch the resource especially during the CRD update. Does it make sense to set up a timeout on deletion and print the manual patch command?

Reply:
We had a discussion offline and decided to open an new issue if the above situation happens.